### DISCUSSION TITLE
no notifications with sailsJs + docker

### DISCUSSION THREAD

**Discussion by atex21** [honeybadger-io/honeybadger-js#824] [Q&A]
Hey!
I'm trying to test honeybadger but it's being impossible.. I'm with sailsJS and docker, configurations as described in documentation:
```
const honeybadgerConfig = {
  apiKey: process.env.HONEYBADGER_API_KEY,
  environment: environment
}
```
variables arrive well to the container in docker.

In a first hand I realized that that configuration about middleware doesn't work.. If I just throw an exception there is no try to send honeybadger notification, the errorHandler is not called =/

In the other hand, if I forgot the last point and use an explicit HoneyBadger.notify, then there is a try to send notification, however I'm stuck with errors coming for no reason =/, namely

Conditions of running | honeybadger-io/js version | Result of a Honeybadger.notify
-- | -- | --
Throw in class + notify in servererror | v3.2.9 | (1)TypeError [ERR_INVALID_ARG_TYPE]: The "listener" argument must be of type function. Received an instance of Object
Throw in class | v3.2.9 | Don’t notify / don’t arrive to errorHandler
Throw in class + notify in servererror | V4.0.3 | (2)TypeError: Cannot read property 'breadcrumbs' of undefined
Throw in class | V4.0.3 | Don’t notify / don’t arrive to errorHandler

a bit more detail:

```
(1)
[API] TypeError [ERR_INVALID_ARG_TYPE]: The "listener" argument must be of type function. Received an instance of Object
[API]     at checkListener (events.js:135:11)
[API]     at ClientRequest.once (events.js:539:3)
[API]     at new ClientRequest (_http_client.js:215:10)
[API]     at Object.request (https.js:374:10)
[API]     at Object.wrappedRequest (/usr/src/app/node_modules/newrelic/lib/instrumentation/core/http.js:412:22)
[API]     at Object.<anonymous> (/usr/src/app/node_modules/agent-base/patch-core.js:25:22)
[API]     at Object.<anonymous> (/usr/src/app/node_modules/socks-proxy-agent/node_modules/agent-base/patch-core.js:23:20)
[API]     at Object.<anonymous> (/usr/src/app/node_modules/agent-base/patch-core.js:25:22)
[API]     at Object.<anonymous> (/usr/src/app/node_modules/socks-proxy-agent/node_modules/agent-base/patch-core.js:23:20)
[API]     at Object.<anonymous> (/usr/src/app/node_modules/agent-base/patch-core.js:25:22)
[API]     at Object.request (/usr/src/app/node_modules/socks-proxy-agent/node_modules/agent-base/patch-core.js:23:20)
[API]     at /usr/src/app/node_modules/@honeybadger-io/js/dist/server/honeybadger.js:932:33
[API]     at parseStats (/usr/src/app/node_modules/@honeybadger-io/js/dist/server/honeybadger.js:757:16)
[API]     at FSReqCallback.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:73:3)
[API]     at FSReqCallback.callbackTrampoline (internal/async_hooks.js:131:17)
```

```
(2)
[API] /usr/src/app/node_modules/@honeybadger-io/js/dist/server/honeybadger.js:871
[API]         var breadcrumbs = store.breadcrumbs;
[API]                                 ^
[API] 
[API] TypeError: Cannot read property 'breadcrumbs' of undefined
[API]     at Honeybadger.Client.addBreadcrumb (/usr/src/app/node_modules/@honeybadger-io/js/dist/server/honeybadger.js:871:33)
[API]     at Honeybadger.Client.notify (/usr/src/app/node_modules/@honeybadger-io/js/dist/server/honeybadger.js:727:14)
[API]     at process.honeybadgerUncaughtExceptionListener (/usr/src/app/node_modules/@honeybadger-io/js/dist/server/honeybadger.js:1151:28)
[API]     at process.emit (events.js:412:35)
[API]     at process.wrappedEmit (/usr/src/app/node_modules/newrelic/lib/instrumentation/core/globals.js:52:23)
[API]     at process.emit (/usr/src/app/node_modules/ts-node/node_modules/source-map-support/source-map-support.js:516:21)
[API]     at process.<anonymous> (internal/process/execution.js:167:25)
[API]     at process.wrappedFatalException [as _fatalException] (/usr/src/app/node_modules/newrelic/lib/instrumentation/core/globals.js:32:23)
[API] [nodemon] app crashed - waiting for file changes before starting...

```
Can you help please?

Thank you in advance.

**Comment by subzero10** [comment]
Hey @atex21, thanks for submitting an issue. I just tried setting up sails.js on a sample project and I ran into some issues as well. However, I was able to manually report errors with `Honeybadger.notify` 🤔 .

By looking at the error above, I am wondering if the issue could be related to your node.js version, because `store` is either `AsyncLocalStorage` or a custom polyfill we wrote in case it is not available. Can you please share your node.js version?

@shalvah Can you also please take a look at this? I have no experience with the sails.js framework.

Note: Created issue #825 to track this better.

**Reply by atex21** [reply]
Hi @subzero10, thank you for answering!

The node used is v14.17.6 and sails is 1.5.2.
I also tried with notifyAsync as mentioned in the created issue, but the issue was the same 😕
